pkg/application/client.go

```diff
@@ -63,10 +67,381 @@ func (c *ApplicationClient) List(ctx context.Context, limit, offset int) (*Appli
+    if err != nil {
+        return nil, fmt.Errorf("failed to find application: %w", err)
+    }
+    
+    if resp.StatusCode() != http.StatusOK {
+        return nil, fmt.Errorf("failed to find application: status %d, body: %s", 
+            resp.StatusCode(), string(resp.Body))
+    }
+    
+    if resp.JSON200 == nil || resp.JSON200.Applications == nil || len(*resp.JSON200.Applications) == 0 {
+        return nil, fmt.Errorf("application with name '%s' not found", name)
+    }
+    
+    var targetApp *ApplicationListItem
+    for _, app := range *resp.JSON200.Applications {
+		if app.Name != nil && *app.Name == name {
+            targetApp = &app
+            break
+        }
+    }
+    
+    if targetApp == nil {
+        return nil, fmt.Errorf("application with name '%s' not found", name)
+    }
+    
+    return c.getApplicationDetails(ctx, *targetApp.Id)
+}
+
+func (c *ApplicationClient) buildSPARequest(name, redirectURL string) (ApplicationModel, error) {
+	allowedOrigins, err := extractOrigins(redirectURL)
+	if err != nil {
+		return ApplicationModel{}, err
+	}
+
+	defaultAuthenticationSequenceType := DEFAULT
+	defaultClaimDialect := LOCAL
+
+	return ApplicationModel{
+		Name: name,
+		AdvancedConfigurations: &AdvancedApplicationConfiguration{
+			DiscoverableByEndUsers: boolPtr(false),
+			SkipLogoutConsent:      boolPtr(true),
+			SkipLoginConsent:       boolPtr(true),
+		},
+		AuthenticationSequence: &AuthenticationSequence{
+			Type: &defaultAuthenticationSequenceType,
+			Steps: &[]AuthenticationStepModel{
+				{
+					Id: 1,
+					Options: []Authenticator{
+						{
+							Idp:           "LOCAL",
+							Authenticator: "basic",
+						},
+					},
+				},
+			},
+		},
+		ClaimConfiguration: &ClaimConfiguration{
+			Dialect: &defaultClaimDialect,
+			RequestedClaims: &[]RequestedClaimConfiguration{
+				{
+					Claim: Claim{
+						Uri: "http://wso2.org/claims/username",
+					},
+				},
+			},
+		},
+		InboundProtocolConfiguration: &InboundProtocols{
+			Oidc: &OpenIDConnectConfiguration{
+				AccessToken: &AccessTokenConfiguration{
+					ApplicationAccessTokenExpiryInSeconds: int64Ptr(3600),
+					BindingType:                           stringPtr("sso-session"),
+					RevokeTokensWhenIDPSessionTerminated:  boolPtr(true),
+					Type:                                  stringPtr("JWT"),
+					UserAccessTokenExpiryInSeconds:        int64Ptr(3600),
+					ValidateTokenBinding:                  boolPtr(false),
+				},
+				GrantTypes:     []string{"authorization_code", "refresh_token"},
+				AllowedOrigins: &allowedOrigins,
+				CallbackURLs:   &[]string{redirectURL},
+				Pkce: &OAuth2PKCEConfiguration{
+					Mandatory:                      boolPtr(true),
+					SupportPlainTransformAlgorithm: boolPtr(false),
+				},
+				PublicClient: boolPtr(true),
+				RefreshToken: &RefreshTokenConfiguration{
+					ExpiryInSeconds:   int64Ptr(86400),
+					RenewRefreshToken: boolPtr(true),
```

Comment:
Shouldn't we at least let the expiary time to be customisable, can leave these as default values

Reply:
This needs to be provided over an update API and a generic API with which you can create any app type

Reply:
Resolving based on above